### PR TITLE
Exception messages were moved to resource file

### LINKED ISSUE #328: Extract error messages from Exceptions to resources
This is subtask of #131.

**Exit criteria:**
1. Error messages from exceptions are localizable (in resource file)

### PR DESCRIPTION
**Pull request by MichalPavlik**
Fixes #328

### CODE REVIEW

**Review by baronfel — APPROVED** [review]
:shipit:

File: Microsoft.NET.Build.Containers/Resources/Strings.resx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <data name="AmazonRegistryFailed" xml:space="preserve">
+    <value>Request to Amazon Elastic Container Registry failed prematurely. This is often caused when the target repository does not exist in the registry.</value>
+  </data>
+  <data name="BlobUploadFailed" xml:space="preserve">
+    <value>Failed to upload blob to {0}; received {1} with detail {2}</value>
+  </data>
+  <data name="CouldntDeserializeJsonToken" xml:space="preserve">
+    <value>Could not deserialize token from JSON</value>
```
Comment (vlada-shubina) [inline Microsoft.NET.Build.Containers/Resources/Strings.resx:127]: Exception message should end with period: https://learn.microsoft.com/en-us/dotnet/api/system.exception.message?view=net-7.0#remarks

Comment (MichalPavlik) [reply Microsoft.NET.Build.Containers/Resources/Strings.resx:127]: Fixed in #351

File: Microsoft.NET.Build.Containers/LocalDaemons/LocalDocker.cs [hunk 2]
```diff
@@ -78,9 +79,12 @@ private static async Task<JsonDocument> GetConfig()
             RedirectStandardError = true
         };
         var proc = Process.Start(psi);
-        if (proc is null) throw new Exception("Failed to start docker client process");
+        if (proc is null) throw new Exception(Resource.GetString(nameof(Strings.DockerProcessCreationFailed)));
         await proc.WaitForExitAsync().ConfigureAwait(false);
-        if (proc.ExitCode != 0) throw new Exception($"Failed to get docker info({proc.ExitCode})\n{await proc.StandardOutput.ReadToEndAsync().ConfigureAwait(false)}\n{await proc.StandardError.ReadToEndAsync().ConfigureAwait(false)}");
+        if (proc.ExitCode != 0) throw new Exception(Resource.FormatString(
```
[the remark this answers is not in the document]
Comment (MichalPavlik) [reply Microsoft.NET.Build.Containers/LocalDaemons/LocalDocker.cs:84]: Fixed in #351

**Pull request merged**